src/hg_connect_stdio.rs

```diff
@@ -186,7 +186,8 @@ pub fn get_stdio_connection(url: &Url, flags: c_int) -> Option<Box<dyn HgRepo>>
         .port()
         .map(|port| CString::new(port.to_string()).unwrap());
     let path = if url.scheme() == "ssh" {
-        percent_decode_str(url.path().trim_start_matches('/')).collect_vec()
+        let path = url.path();
+        percent_decode_str(path.strip_prefix('/').unwrap_or(path)).collect_vec()
```

Comment:
This likely could be simplified to `url.path().strip_prefix('/').unwrap()`.
With all the checks made above this function, I can't imagine the situation when the path doesn't start with `/`.